### PR TITLE
GFORMS-1761 - Fix rounding issue when 2 rounded top level expressions…

### PR DESCRIPTION
**Pull request by DursunGundogan**
… are added together

### CODE REVIEW

File: app/uk/gov/hmrc/gform/eval/EvaluationResults.scala [hunk 1]
```diff
@@ -238,7 +238,12 @@ case class EvaluationResults(
   def evalTyped(er: ExpressionResult, tpe: ExplicitExprType): ExpressionResult =
     tpe match {
       case ExplicitExprType.Sterling(roundingMode) =>
-        er.withNumberResult(bigDecimal => NumberSetScale.setScale(bigDecimal, 2, roundingMode))
+        er match {
+          case NumberResult(_) =>
+            er.withNumberResult(bigDecimal => NumberSetScale.setScale(bigDecimal, 2, roundingMode))
+          case _ => er
+        }
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/eval/EvaluationResults.scala:245]: How is this not the same as 
```
er.withNumberResult(bigDecimal => NumberSetScale.setScale(bigDecimal, 2, roundingMode))
```
? 

Can you provide minimal example where this makes a difference?

Comment (DursunGundogan) [reply app/uk/gov/hmrc/gform/eval/EvaluationResults.scala:245]: I think we should round only `NumberResult`, not `ListResult`

Comment (DursunGundogan) [reply app/uk/gov/hmrc/gform/eval/EvaluationResults.scala]: Reverted

File: app/uk/gov/hmrc/gform/eval/EvaluationResults.scala [hunk 2]
```diff
@@ -203,8 +203,15 @@ case class EvaluationResults(
...
-          case _               => unsupportedOperation("Number")(expr)
+          case lrs: ListResult =>
+            lrs.list.fold(NumberResult(0)) { case (a, b) =>
+              val roundedB = field1 match {
+                case Typed(_, tpe) => evalTyped(b, tpe)
+                case _             => b
+              }
+              a + roundedB
+            }
+          case _ => unsupportedOperation("Number")(expr)
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/eval/EvaluationResults.scala:214]: This is weird. Why is rounding logic living in the branch of code where we are adding numbers together? This is not systematic solution to the problem.

**Pull request closed without merging**